nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -539,204 +789,306 @@ in
+            config.environment.etc."ssh/sshd_config".source
+          ];
+
+          preStart = ''
+            # Make sure we don't write to stdout, since in case of
+            # socket activation, it goes to the remote side (#19589).
+            exec >&2
+
+            ${lib.flip lib.concatMapStrings cfg.hostKeys (k: ''
+              if ! [ -s "${k.path}" ]; then
+                  if ! [ -h "${k.path}" ]; then
+                      rm -f "${k.path}"
+                  fi
+                  mkdir -p "$(dirname '${k.path}')"
+                  chmod 0755 "$(dirname '${k.path}')"
+                  ssh-keygen \
+                    -t "${k.type}" \
+                    ${lib.optionalString (k ? bits) "-b ${toString k.bits}"} \
+                    ${lib.optionalString (k ? rounds) "-a ${toString k.rounds}"} \
+                    ${lib.optionalString (k ? comment) "-C '${k.comment}'"} \
+                    ${lib.optionalString (k ? openSSHFormat && k.openSSHFormat) "-o"} \
+                    -f "${k.path}" \
+                    -N ""
+              fi
+            '')}
+          '';
+
+          serviceConfig =
+            {
+              CapabilityBoundingSet = [
+                "~CAP_AUDIT_CONTROL"
+                "~CAP_AUDIT_READ"
+                "~CAP_AUDIT_WRITE"
+                "~CAP_BLOCK_SUSPEND"
+                "~CAP_BPF"
+                "~CAP_CHOWN"
+                "~CAP_DAC_OVERRIDE"
+                "~CAP_DAC_READ_SEARCH"
+                "~CAP_FOWNER"
+                "~CAP_FSETID"
+                "~CAP_IPC_LOCK"
+                "~CAP_IPC_OWNER"
+                "~CAP_KILL"
+                "~CAP_LEASE"
+                "~CAP_LINUX_IMMUTABLE"
+                "~CAP_MAC_ADMIN"
+                "~CAP_MAC_OVERRIDE"
+                "~CAP_MKNOD"
+                "~CAP_NET_ADMIN"
+                "~CAP_NET_BROADCAST"
+                "~CAP_NET_RAW"
+                "~CAP_SETFCAP"
+                "~CAP_SETPCAP"
+                "~CAP_SYSLOG"
+                "~CAP_SYS_ADMIN"
+                "~CAP_SYS_BOOT"
+                "~CAP_SYS_MODULE"
+                "~CAP_SYS_NICE"
+                "~CAP_SYS_PACCT"
+                "~CAP_SYS_PTRACE"
+                "~CAP_SYS_RAWIO"
+                "~CAP_SYS_RESOURCE"
+                "~CAP_SYS_TIME"
+                "~CAP_SYS_TTY_CONFIG"
+                "~CAP_WAKE_ALARM"
+              ];
+              ExecStart =
+                (lib.optionalString cfg.startWhenNeeded "-")
+                + "${cfg.package}/bin/sshd "
+                + (lib.optionalString cfg.startWhenNeeded "-i ")
+                + "-D "
+                # don't detach into a daemon process
+                + "-f /etc/ssh/sshd_config";
+              KillMode = "process";
+              LockPersonality = true;
+              RestrictAddressFamilies = [
+                "~AF_NETLINK"
+                "~AF_PACKET"
+              ];
+              RestrictNamespaces = [
```

Comment:
Why not use just true?

Reply:
1. Because I'm following the recommendations of `systemd-analyze security sshd`.
2. Because this might break something. If it does break something, reverting a single line commit is going to be miles easier than figuring out what all the namespaces are, and then excluding one of them.